vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -150,10 +161,17 @@ void AdsImpl::OnPageLoaded(
     last_visible_tab_url = last_visible_tab->url;
   }
 
-  purchase_intent_classifier_->MaybeExtractIntentSignal(url,
-      last_visible_tab_url);
+  if (!SameDomainOrHost(url, last_visible_tab_url)) {
```

Comment:
Not for now, but I'm on the fence again whether this piece of business logic should be responsibility of `AdsImpl` or the processor. Maybe a processor should observe tab updates to decide internally what action to take on subsequent visits for the same eTLD+1.

Reply:
Agreed and we need to think of a better solution as right now a user can visit site a, change to site b and return to site a. Where site a would then be extracted again

Reply:
moved to backlog since not for now